src/index.js

```diff
@@ -27,22 +29,35 @@ class ZenfiSDK {
   }
 
   async fetchData() {
-    if (this.leadData) return self;
-    // Fetch data from Zenfi API
-    // Set results to leadData
+    if (!this.leadInfo) this.leadInfo = await fetchLeadInfo(this.token);
+    return this.leadInfo;
   }
 
-  fillForm() {
-    this.fetchData();
-    // Iterate all targets
+  async fillTargets() {
+    await this.fetchData();
+    (this.targets || []).forEach(({ dataKey, selector, strategy }) => {
+      const value = (this.leadInfo || {})[dataKey];
+      if (value) fillTarget({ selector, value, strategy });
+    });
```

Comment:
```suggestion
    if(!Array.isArray(this.targets) || !this.leadInfo) return;
    
    this.targets.forEach(({ dataKey, selector, strategy }) => {
      const value = this.leadInfo[dataKey];
      if (value) fillTarget({ selector, value, strategy });
    });
```

Esto me parece un poco más fácil de procesar que los `|| []`